crates/fuel-core/src/database/metadata.rs

```diff
@@ -1,127 +1,109 @@
 use crate::{
     database::{
+        database_description::{
+            DatabaseDescription,
+            DatabaseMetadata,
+        },
         storage::UseStructuredImplementation,
-        Column,
         Database,
         Error as DatabaseError,
     },
     state::DataSource,
 };
-use fuel_core_chain_config::ChainConfig;
 use fuel_core_storage::{
     blueprint::plain::Plain,
     codec::postcard::Postcard,
+    not_found,
     structured_storage::{
         StructuredStorage,
         TableWithBlueprint,
     },
+    Error as StorageError,
     Mappable,
     Result as StorageResult,
-    StorageMutate,
+    StorageAsRef,
 };
+use fuel_core_types::fuel_merkle::storage::StorageMutate;
 
-/// The table that stores all metadata. Each key is a string, while the value depends on the context.
-/// The tables mostly used to store metadata for correct work of the `fuel-core`.
-pub struct MetadataTable<V>(core::marker::PhantomData<V>);
+/// The table that stores all metadata about the database.
+pub struct MetadataTable<Description>(core::marker::PhantomData<Description>);
 
-impl<V> Mappable for MetadataTable<V>
+impl<Description> Mappable for MetadataTable<Description>
 where
-    V: Clone,
+    Description: DatabaseDescription,
 {
-    type Key = str;
-    type OwnedKey = String;
-    type Value = V;
-    type OwnedValue = V;
+    type Key = ();
```

Comment:
This feels like a smell to me. Does this need to be `Mappable`?

Reply:
My idea was to have a compilation know metadata types instead of `(string, string)` pairs.

Since each database only has one metadata entry, we don't need something else than `()`=) But it works worse in the case of the huge metadata entry since it is slower to serialize/deserialize.

I'm open to any suggestions=) I only tried to make it more safe(type defined).

Reply:
It's not the worst thing. Just seems like not everything needs to be `Mappable`. I prefer this over stringly-typing things for sure :+1: